xla/stream_executor/cuda/cuda_blas_lt.h

```diff
@@ -22,21 +22,36 @@ limitations under the License.
 #include <string>
 #include <vector>
 
-#include "third_party/gpus/cuda/include/cublasLt.h"
-#include "third_party/gpus/cuda/include/cublas_v2.h"
-#include "third_party/gpus/cuda/include/cuda.h"
 #include "xla/stream_executor/blas.h"
-#include "xla/stream_executor/cuda/cuda_blas_utils.h"
 #include "xla/stream_executor/device_memory.h"
 #include "xla/stream_executor/host_or_device_scalar.h"
 #include "tsl/platform/status.h"
 
+#ifndef TENSORFLOW_USE_ROCM
+#include "third_party/gpus/cuda/include/cublasLt.h"
+#include "third_party/gpus/cuda/include/cublas_v2.h"
+#include "third_party/gpus/cuda/include/cuda.h"
+#include "xla/stream_executor/cuda/cuda_blas_utils.h"
+#else
+#include "rocm/rocm_config.h"
+#include "../rocm/hipblaslt_wrapper.h"
```

Comment:
To me this looks like an indication that it would be better to have this file one directory up, and name it gpu_blas_lt.h
Also, the implementation file likely should be one directory up, and cuda and rocm implementation could be subclasses that have specific virtual methods for where they differ.

Reply:
Yes, it's much better in this way.

Reply:
Hi @akuegel I have fixed CI checks on cuda side, wonder would be possible to get merged first and I will send another PR to redesign this part? as this hipblas_lt is urgent and important to us due to JAX.

As a side note, Check Contents / check-contents keeps failed for tsl::Status, but I can see other merged PRs are ok with tsl::Status(), e.g., https://github.com/openxla/xla/commit/386fd8513ea7430a527e4917e28fb6f1262c0d4b

Reply:
I would be ok if you just copy the code for now, and later work on unifying it with a shared base class. But please don't add rocm specific code inside the cuda directory.

Reply:
Thanks for the advice. Initially we think this is the leastest code change to achieve....Yes, we wouldn't do it again.

Reply:
Hi @ddunl wonder will be ok to merge this PR first? as it's urgent and important to use having JAX to use hipblas_lt.

@akuegel is ok for now and I will create another PR (it's on progress) to redesign and unify hipblas_lt and cublas_lt.

Reply:
I think I'd have trouble getting this merged with the rocm references under stream_executor/cuda, but if you can fix that and have tests passing I can try to merge it later today

Reply:
Thanks a lot! @ddunl  

It's already passed all tests in CI and it's #ifndef TENSORFLOW_USE_ROCM, so it wouldn't call any rocm references in cuda side.

Reply:
I mean even if it's ifdef'd out, I should be able to do `grep -R rocm` under `stream_executor/cuda` and get 0 results, that's what I think I'll have trouble merging if I import it

Reply:
Ok, I will try to copy past and move it out first. Thanks for clarification!

Reply:
Thanks @ddunl for explaining what I meant :)
Yes, we also don't want any ROCM specific #ifdefs in stream_executor/cuda. Copying the code and adjusting it is also not great, but still better. And then you can work on unifying the code with a common base class later in a separate PR.

Reply:
Would it satisfy you if we moved the existing cuda_blas_lt.cc entirely into stream_executor/gpu instead?